File: Other/888.Fair Candy Swap/Solution.cpp
class Solution {
public:
    vector<int> fairCandySwap(vector<int> A, vector<int> B) {
        int dif = (accumulate(A.begin(), A.end(), 0) - accumulate(B.begin(), B.end(), 0)) / 2;

        unordered_set<int> S(A.begin(), A.end());

        for (int b: B)
            if (S.count(b + dif))
                return {b + dif, b};
    }
};

File: Other/888.Fair Candy Swap/Solution.py
class Solution(object):
    def fairCandySwap(self, A, B):
        """
        :type A: List[int]
        :type B: List[int]
        :rtype: List[int]
        """
        dif = (sum(A) - sum(B)) / 2
        A = set(A)
        for b in set(B):
            if dif + b in A:
                return [dif + b, b]